dotCMS/src/main/java/com/dotcms/services/VanityUrlServices.java

```diff
@@ -0,0 +1,189 @@
+    }
+
+    /**
+     * Remove all the vanity Url contentlet versions from the vanity URL cache
+     *
+     * @param vanityUrl The vanity URL contentlet object
+     */
+    public void invalidateAllVanityUrlVersions(Contentlet vanityUrl) {
+        try {
+            Identifier identifier = identifierAPI.find(vanityUrl.getIdentifier());
+            List<Contentlet> contentletVersions = contentletAPI
+                    .findAllVersions(identifier, APILocator.systemUser(), false);
+            contentletVersions.stream()
+                    .filter(con -> vanityUrl.getLanguageId() == con.getLanguageId())
+                    .forEach((Contentlet con) -> removeFromCache(con));
+
+            removeFromCache(vanityUrl);
+        } catch (DotDataException | DotRuntimeException | DotSecurityException e) {
+            Logger.error(VanityUrlServices.class,
+                    "Error trying to invalidate Vanity URL identifier:" + vanityUrl.getIdentifier(),
+                    e);
+        }
+    }
+
+    /**
+     * Remove the vanity Url contentlet version from cache
+     */
+    private void removeFromCache(Contentlet contentlet) {
+        try {
+            vanityURLCache.remove(VanityUrlUtil.sanitizeKey(contentlet));
+            vanityURLCache
+                    .removeCachedVanityUrls(VanityUrlUtil.sanitizeSecondCachedKey(contentlet));
+        } catch (DotDataException | DotRuntimeException | DotSecurityException e) {
+            Logger.error(VanityUrlServices.class,
+                    "Error trying to invalidate Vanity URL identifier:" + contentlet
+                            .getIdentifier(), e);
+        }
+    }
+
+    /**
+     * Load in cache the active vanities Urls
+     */
+    public void initializeVanityUrlCache() {
+        List<CachedVanityUrl> activeVanityUrls = APILocator.getVanityUrlAPI()
+                .getActiveCachedVanityUrls(APILocator.systemUser());
+        activeVanityUrls.stream().forEach((CachedVanityUrl vanity) -> {
+            Set<CachedVanityUrl> currentCachedVanities = CacheLocator.getVanityURLCache()
+                    .getCachedVanityUrls(vanity.getSiteId());
+            currentCachedVanities.add(vanity);
+            CacheLocator.getVanityURLCache().setCachedVanityUrls(VanityUrlUtil
+                            .sanitizeSecondCacheKey(vanity.getSiteId(), vanity.getLanguageId()),
+                    currentCachedVanities);
+        });
+    }
+
+    /**
+     * Load in cache the active vanities Urls
+     * searching by host and languageId
+     *
+     * @param hostId The current host Id
+     * @param languageId The current language Id
+     */
+    public void initializeVanityUrlCache(String hostId, long languageId) {
+        List<VanityUrl> activeVanityUrls = APILocator.getVanityUrlAPI()
+                .getActiveVanityUrlsByHostAndLanguage(hostId, languageId, APILocator.systemUser());
+        activeVanityUrls.stream().forEach((VanityUrl vanity) -> updateCache(vanity));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)